api/src/test/java/org/apache/iceberg/expressions/TestAggregateEvaluator.java

```diff
@@ -0,0 +1,184 @@
+          Row.of(),
+          50,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 50L, 3, 50L, 4, 50L),
+          // null value counts
+          ImmutableMap.of(1, 10L, 3, 50L, 4, 10L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 2345)));
+
+  private static final DataFile MISSING_SOME_NULLS_STATS_1 =
+      new TestDataFile(
+          "file_2.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 0L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 100)));
+
+  private static final DataFile MISSING_SOME_NULLS_STATS_2 =
+      new TestDataFile(
+          "file_3.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), -33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 3333)));
+
+  private static final DataFile[] dataFiles = {
+    FILE, MISSING_SOME_NULLS_STATS_1, MISSING_SOME_NULLS_STATS_2
+  };
+
+  @Test
+  public void testIntAggregate() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.countStar(),
+            Expressions.count("id"),
+            Expressions.max("id"),
+            Expressions.min("id"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    Assert.assertTrue(aggregateEvaluator.allAggregatorsValid());
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {90L, 60L, 3333, -33};
+    assertEvaluatorResult(result, expected);
+  }
+
+  @Test
+  public void testAllNulls() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.countStar(),
+            Expressions.count("all_nulls"),
+            Expressions.max("all_nulls"),
+            Expressions.min("all_nulls"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    Assert.assertTrue(aggregateEvaluator.allAggregatorsValid());
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {90L, 0L, null, null};
```

Comment:
`count(*)` has values for all the files. It's 90L

`count("all_nulls")` has value for all the FILEs, 1st FILE, count is (50 - 50), second FILE, count is (20 - 20), third FILE, count is (20 - 20), so count is 0.
 
`max(("all_nulls")` has value for all the FILEs, for all the FILES (hasUpperBound == false but boundAllNull == true), the value is null

`min(("all_nulls")` has value for all the FILEs, for all the FILES (hasLowerBound == false but boundAllNull == true), the value is null